go/pkg/pwagent/nginx.go

```diff
@@ -250,8 +254,20 @@ func Nginx(ctx context.Context, opts AgentOpts, cli *client.Client, logger *zap.
 
 					// add hash per users to proxy configuration
 					if _, found := opts.AllowedUsers[instance.Names[0][1:]]; found {
-						for _, hash := range opts.AllowedUsers[instance.Names[0][1:]] {
-							upstream.Hashes = append(upstream.Hashes, strconv.Itoa(int(hash)))
+						for _, userID := range opts.AllowedUsers[instance.Names[0][1:]] {
+							stringToHash := fmt.Sprintf("%s%d%s", instance.ID, userID, opts.Salt)
+							hashBytes := make([]byte, 8)
+							hasher := sha3.NewCShake256([]byte("NAME"), []byte("Partition1"))
```

Comment:
```suggestion
							hasher := sha3.NewShake256()
```

Not sure here, what did you expect by using custom N and S for ShakeHash?